enclave-runtime/src/top_pool/top_pool_container.rs

```diff
@@ -0,0 +1,111 @@
+/*
+	Copyright 2021 Integritee AG and Supercomputing Systems AG
+
+	Licensed under the Apache License, Version 2.0 (the "License");
+	you may not use this file except in compliance with the License.
+	You may obtain a copy of the License at
+
+		http://www.apache.org/licenses/LICENSE-2.0
+
+	Unless required by applicable law or agreed to in writing, software
+	distributed under the License is distributed on an "AS IS" BASIS,
+	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+	See the License for the specific language governing permissions and
+	limitations under the License.
+
+*/
+
+use crate::top_pool::{pool_types::BPool, primitives::TrustedOperationPool};
+use sp_core::H256;
+use std::{
+	marker::PhantomData,
+	sync::{
+		atomic::{AtomicPtr, Ordering},
+		Arc, SgxMutex,
+	},
+};
+
+static GLOBAL_TOP_CONTAINER: AtomicContainer = AtomicContainer::new();
+
+/// getter trait for the trusted operation pool
+pub trait GetTopPool: Send + Sync + 'static {
+	type TrustedOperationPool: TrustedOperationPool<Hash = H256> + 'static;
```

Comment:
This trait bound was essential (`<Hash = H256>`), thanks @clangenb for noticing that! It seems we have parts in our code that are not well abstracted yet, and in order to interface with those parts, we need to restrict the Hash type here.

Reply:
Yes, this has been bugging me to such a great extent that I'd like to tackle this issue after I finish my current task.